pallets/pools/src/lib.rs

```diff
@@ -1250,32 +1402,56 @@ pub mod pallet {
 					pool.last_epoch_executed,
 					tranche,
 					solution,
-					execution,
+					executed_amounts,
 					epoch_tranche.price,
 				)?;
 			}
 
 			// Update the total/available reserve for the new total value of the pool
-			pool.total_reserve = pool
-				.total_reserve
-				.checked_add(&total_supply)
-				.and_then(|res| res.checked_sub(&total_redeem))
+			pool.total_reserve = executed_amounts
+				.iter()
+				.fold(
```

Comment:
With a low reserve, any single tranche having more redemptions than investments could cause this to underflow now and we'd flag an error. That seems more likely than the total of investments or the total of redemptions individually overflowing, and is why I precalculated those totals before.